src/Media/Audio/OpenALSoundProvider.cpp

```diff
@@ -733,7 +841,9 @@ bool AudioSample16::Open(PAudioDataSource data_source) {
         return false;
     }
 
-    if (!pDataSource->Open()) {
+    std::shared_ptr<OpenALAudioDataSource> openalDataSource = std::dynamic_pointer_cast<OpenALAudioDataSource, IAudioDataSource>(pDataSource);
```

Comment:
You're relying on this one is always `OpenALAudioDataSource`. Just take this type as a function parameter and get rid of the cast.